pypeit/edgetrace.py

```diff
@@ -4076,11 +4098,26 @@ def maskdesign_matching(self, debug=False):
                 omodel_bspat[i] = omodel_tspat[i]
                 omodel_tspat[i] = invert_order
 
+        # If there are overlapping slits, i.e., omodel_bspat[i+1]<omodel_tspat[i], switch those
+        for i in range(sortindx.size -1):
+            if omodel_tspat[sortindx][i] != -1 and omodel_bspat[sortindx][i+1] != -1 and \
+                    omodel_tspat[sortindx][i] > omodel_bspat[sortindx][i+1]:
+                tmt_bspat = omodel_tspat[sortindx][i]
+                omodel_tspat[sortindx[i]] = omodel_bspat[sortindx[i+1]]
+                omodel_bspat[sortindx[i+1]] = tmt_bspat
+                # Re-check If the `omodel_bspat` is greater than `omodel_tspat` and switch the order.
+                # It may happens if 3 slits are overlapping (true story!)
```

Comment:
Seems we might need a `while` or `for` loop here instead of nested if/then
in case there are 4!   :) 

But I'm ok with this for now.

Reply:
Sorry, I wasn't able to find another way to code this part.

Reply:
No worries.

Reply:
As long as you're both okay with it, I think it's okay if we punt on this for now.  But...

- Note you can swap values in python without an intermediate variable.  Something like, `x, y = y, x`.

- @profxj , I thought we decided that you wanted the edges of the overlapping slits to instead be the same so that the flexure correction doesn't have gaps in the spatial slit illumination pattern that you use in the cross-correlation.  I don't think this does this, right?

- We probably do need something general that can deal with more than just 3 overlapping slits.  Can we instead check this based on the optical model and then throw an error if that's the case?  That way, we don't need to write the more general code now, but we'll know if the code ever runs across a dataset where this is an issue.

Reply:
Ok. I did change this part of the code. Instead of switching the overlapping traces, what we do now is to move the left and right edges to be adjacent. I tested with the case of 3 overlapping slits and works well there too.